### PR TITLE
Added etcd datastore.

### PR DESCRIPTION
**Pull request by IKermani**

### CODE REVIEW

File: database/etcd.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+//   - key:
+//   - insecure-skip-tls-verify: false
+//   - args:
+type Etcd struct {
+	Base
+	endpoints             []string
+	user                  string
+	password              string
+	caCert                string
+	cert                  string
```
Comment (huacnlee) [inline database/etcd.go:36]: Is the `cert`, `caCert` and `key` is a requirement? 
If not, I'd like to leave this to use `args`, too many options will make GoBackup hard for maintaining and testing.

**Comment by huacnlee** [comment]
Please follow this doc to add a development guides for **etcd**.

https://github.com/gobackup/gobackup/blob/main/docs/development-for-minio.md

File: database/etcd.go (hunk 1 above)
Comment (IKermani) [reply database/etcd.go:36]: No these args are not required, but in most cases of deploying etcd in kubernetes these args are required.
Also neither username nor password are required arguments for etcd.

File: database/etcd.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+//   - cacert:
+//   - cert:
+//   - key:
+//   - insecure-skip-tls-verify: false
+//   - args:
+type Etcd struct {
+	Base
+	endpoints     []string
+	args          string
+	_dumpFilePath string
```
Comment (huacnlee) [inline database/etcd.go:33]: Em...

Sorry, again, why do you removed the `username` and `password`.

You can follow this config.

- endpoints
- username
- password
- skip_verify 

https://gobackup.github.io/configuration/databases/influxdb

Comment (huacnlee) [reply database/etcd.go:33]: Or is the `user` and `password` is also optional in ectd?

Comment (IKermani) [reply database/etcd.go:33]: You can choose whether to enable authentication in etcd. By default, it's disabled, and anyone can access the data. If you don't enable authentication, username and password become irrelevant.

Comment (huacnlee) [reply database/etcd.go:33]: Obey, I got it.

**Comment by IKermani** [comment]
Would you please release a new version?

**Comment by huacnlee** [comment]
https://github.com/gobackup/gobackup/releases/tag/v2.10.0

**Pull request merged**